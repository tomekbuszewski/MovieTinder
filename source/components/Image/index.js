import React from 'react';
import PropTypes from 'prop-types';

/**
 * Image component
 * @param {string} src
 * @param {string} alt
 * @param {string|bool} title
 * @param {any} className
 * @returns {*}
 * @constructor
 */
// @flow
const Image = ({ src, alt, title, className }: { src: string, alt: string, title: string | boolean, className: any }) =>
<img src={src} alt={alt} title={title} className={className} />;

Image.propTypes = {
  src: PropTypes.string,
  alt: PropTypes.string,
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.bool])
};

Image.defaultProps = {
  title: false,
  alt: 'Image',
  className: ''
};

export default Image;
