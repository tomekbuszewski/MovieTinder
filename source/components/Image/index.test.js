import React from 'react';
import renderer from 'react-test-renderer';

import Image from './index';

describe ('Image tests', () => {
  it ('should match snapshot with all props', () => {
    const tree = renderer.create(<Image src="https://some.image" alt="Test alt" title="Test title" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it ('should match snapshot without title', () => {
    const tree = renderer.create(<Image src="https://some.image" alt="Test alt" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
