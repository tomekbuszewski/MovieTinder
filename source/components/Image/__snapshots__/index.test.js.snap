// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image tests should match snapshot with all props 1`] = `
<img
  alt="Test alt"
  className=""
  src="https://some.image"
  title="Test title"
/>
`;

exports[`Image tests should match snapshot without title 1`] = `
<img
  alt="Test alt"
  className=""
  src="https://some.image"
  title={false}
/>
`;
